src/main/java/com/SAS/teamManagenemt/TeamManagement.java

```diff
@@ -32,14 +32,63 @@ public User addAdditionalTeamOwner(User newTeamOwner, Team team, User nominatedB
             team.addTeamOwner((TeamOwner)newTeamOwner);
             ((TeamOwner) newTeamOwner).setTeam(team);
             ((TeamOwner) newTeamOwner).setNominatedBy((TeamOwner)nominatedBy);
-
-            return newTeamOwner;
         }
 
         else {
             System.out.println("The user is unauthorized to own the team");
-            return null;
         }
+
+        return newTeamOwner;
+    }
+
+    /**
+     * The function receives a user and a team and adds the user as the team manager
+     * TODO: UI will need to popup a window to get approval
+     * @param newTeamManager
+     * @param team
+     * @param nominatedBy
+     * @param approval - a boolean which determines if the manager got privileges from the team owner
+     * @return newTeamManager - the team manager
+     */
+    public User addTeamManager(User newTeamManager, Team team, User nominatedBy, boolean approval){
+        //check if the team already has a manager
+        if(teamHasManager(team)){
+            System.out.println("The team already has a manager, please remove it first to nominate a new manager");
+            return newTeamManager;
+        }
+        //check if the user can manage this team and that the user that nominates him is the team owner
+        if (validateTeamManager(newTeamManager, team) && ownsTeam(team, nominatedBy)){
+            newTeamManager = userController.addRoleToUser(newTeamManager, UserType.TEAM_MANAGER,approval);
+            team.setTeamManager((TeamManager)newTeamManager);
+            ((TeamManager)newTeamManager).setTeam(team);
+            ((TeamManager)newTeamManager).setNominatedBy((TeamOwner)nominatedBy);
+        }
+        else{
+            System.out.println("The user is unauthorized to manage the team");
+        }
+        return newTeamManager;
+    }
+
+    /**
+     * This function checks if the team already has a manager, if so returns true, otherwise returns false
+     * @param team
+     * @return true or false
+     */
+    private boolean teamHasManager(Team team) {
+        return team.getManager() != null;
+    }
+
+    /**
+     * This function checks if the user is not already a team manager or owner
+     * @param newTeamManager
+     * @param team
+     * @return true or false
+     */
+    private boolean validateTeamManager(User newTeamManager, Team team) {
+        if ((newTeamManager instanceof TeamManager || newTeamManager instanceof TeamOwner)){
+            return false;
+        }
+        return true;
     }
 
```

Comment:
cod looks fine, hate all this casting but i guess thsi is the only way of doing it.